fms/models/qwen3.py

```diff
@@ -0,0 +1,701 @@
+# -*- coding: utf-8 -*-
+
+"""
+Part of code to support Qwen3 models
+"""
+
+# pylint: disable=unknown-option-value,protected-access
+# pylint: disable=unused-argument
+# pylint: disable=too-many-instance-attributes
+# pylint: disable=missing-class-docstring
+# pylint: disable=missing-function-docstring
+# pylint: disable=too-many-arguments,too-many-positional-arguments
+
+import logging
+import math
+import re
+from dataclasses import dataclass
+from typing import Any, Dict, Mapping, Optional, Tuple, Unpack
+
+import torch
+from torch import nn
+
+from fms import models
+from fms.distributed.strategy import (
+    DistributedStrategy,
+    NoOpStrategy,
+)
+from fms.modules.attention import (MultiHeadAttention,
+                                   AttentionKwargs,
+                                   get_attention_type)
+
+from fms.modules.feedforward import GatedLinearUnit
+from fms.modules.layernorm import LayerNormParameterized
+from fms.modules.positions import RotaryEmbedding
+from fms.utils import serialization
+from fms.utils.activation import str_to_activation
+from fms.utils.config import ModelConfig
+
+logger = logging.getLogger(__name__)
+
+"""
+======= Mapping =======
+    # These are independent of model (except architecture)
+    inner_dim = config.intermediate_size
+    architecture = "qwen3"
+    01 + config_params["activation_fn"] = config.hidden_act
+    02 + config_params["emb_dim"] = config.hidden_size
+    03 + config_params["max_expected_seq_len"] = config.max_position_embeddings
+    04 + config_params["kvheads"] = config.num_key_value_heads
+    05 + config_params["p_dropout"] = config.attention_dropout
+    06 + config_params["norm_eps"] = config.rms_norm_eps
+    07 + config_params["rope_base"] = config.rope_theta
+    08 + config_params["src_vocab_size"] = config.vocab_size
+    09 + config_params["nheads"] = config.num_attention_heads
+    10 + config_params["nlayers"] = config.num_hidden_layers
+    11 + config_params["hidden_grow_factor"] = inner_dim / config.hidden_size
+    12 + config_params["tie_heads"] = config.tie_word_embeddings
+
+========= config.json -- https://huggingface.co/Qwen/Qwen3-1.7B/blob/main/config.json
+ "architectures": [
+    "Qwen3ForCausalLM"
+  ],
+  "attention_bias": false,
```

Comment:
Make sure your FMS Qwen3 model receives the HF flag:
In _infer_model_configuration, forward:
config_params["attention_bias"] = config.attention_bias
In fms/models/qwen3.py, store it:
self.attention_bias = bool(config.attention_bias)

Pass attention_bias from HF config into FMS config.
Build q/k/v/o (or fused qkv + o) linears with bias=self.attention_bias.
In the adapter, concat q/k/v biases if fused; map o-bias.
Add tests to ensure both bias-on and bias-off paths are correct and produce zero missing keys.